opentracing-api/src/main/java/io/opentracing/Span.java

```diff
@@ -65,25 +65,24 @@
     Span setTag(String key, Number value);
 
     /**
-     * Add a new log event to the Span, accepting an event name string and an optional structured payload argument.
+     * Log key:value pairs to the Span with the current walltime timestamp.
      *
-     * If specified, the payload argument may be of any type and arbitrary size, though implementations are not
-     * required to retain all payload arguments (or even all parts of all payload arguments).
+     * <p>A contrived example:
+     * <pre>{@code
+     span.log(
+         new LogField("size", rpc.size()),  // numeric values
+         new LogField("URI", rpc.URI()),  // String values
+         new LogField("payload", rpc.payload()));  // Object values
+     }</pre>
      *
-     * The timestamp of this log event is the current time.
-     **/
-    Span log(String eventName, /* @Nullable */ Object payload);
-
+     * @param fields One or more LogField instances
+     * @return the Span, for chaining
+     */
+    Span log(LogField<?>... fields);
     /**
-     * Add a new log event to the Span, accepting an event name string and an optional structured payload argument.
-     *
-     * If specified, the payload argument may be of any type and arbitrary size, though implementations are not
-     * required to retain all payload arguments (or even all parts of all payload arguments).
-     *
-     * The timestamp is specified manually here to represent a past log event.
-     * The timestamp in microseconds in UTC time.
-     **/
-    Span log(long timestampMicroseconds, String eventName, /* @Nullable */ Object payload);
+     * Like log(LogField...), but with an explicit timestamp.
+     */
+    Span log(long timestampMicroseconds, LogField<?>... fields);
```

Comment:
I like it because it's consistent with Finish etc. Also, the finish method doesn't accept logs right now.